src/octokit-queries.ts

```diff
@@ -1,35 +1,37 @@
 import * as core from '@actions/core';
 import { GitHub } from '@actions/github/lib/utils';
+import { isString } from './functions';
 
 export const getTeamSlugsForAuthor = async (
   octokit: InstanceType<typeof GitHub>,
   org: string,
-  author: string,
+  username: string,
+  ignoreSlugs: string[] = [],
 ): Promise<string[]> => {
   const { data: allTeams } = await octokit.rest.teams.list({
     org,
   });
-  const teamSlugs = allTeams.map((team) => team.slug);
 
-  const authorsTeamSlugs = [];
+  const authorsTeamSlugs = allTeams.map<Promise<string | undefined>>(async ({ slug }) => {
+    if (ignoreSlugs.includes(slug)) {
+      return;
+    }
 
-  for await (const teamSlug of teamSlugs) {
     try {
       const { data: membership } = await octokit.rest.teams.getMembershipForUserInOrg({
         org,
-        team_slug: teamSlug,
-        username: author,
+        team_slug: slug,
+        username,
       });
 
       if (membership.state === 'active') {
-        authorsTeamSlugs.push(teamSlug);
+        return slug;
       }
     } catch (error) {
       // Octokit query fails when username is not member of a team, see https://octokit.github.io/rest.js/v18
-      core.info(`${author} not a member of ${teamSlug}`);
-      continue;
+      core.info(`${username} not a member of ${slug}`);
     }
-  }
+  });
 
-  return authorsTeamSlugs;
+  return (await Promise.all(authorsTeamSlugs)).filter(isString);
```

Comment:
Why is `.filter(isString)` necessary here?

Reply:
This filters out the `undefined` values from the mapped array.